src/test/java/updated/game/abstractClasses/Board.java

```diff
@@ -0,0 +1,150 @@
+package updated.game.abstractClasses;
+
+import lombok.Getter;
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+@Getter
+public abstract class Board {
+    public static  int BOARD_SIZE = 3;
```

Comment:
BOARD_SIZE can be entered by user so that it can be extensible when you want to implement nXn TicTcToe

Reply:
I was thinking, as it's an abstract class so let me keep 3 by default, then i would have a separate implementation of class where user is given choice to enter the size of board he/she is willing to play.

Reply:
Keeping the default BOARD_SIZE in the abstract class is fine for a base implementation. However, making it configurable in the derived class where the user can choose their board size would make the game more flexible and align with object-oriented principles.